### PR TITLE
Add condition list command

### PR DESCRIPTION
**Pull request by 16yuki0702**
# Changes
Add condition subcommand and feature of listing its resources https://github.com/tektoncd/cli/issues/286

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
you can list condition resources from tkn
```

### CODE REVIEW

File: pkg/cmd/condition/list_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+			command:   []string{"ls", "-n", "ns"},
+			input:     seeds[0],
+			wantError: false,
+			want: []string{
+				"NAME         AGE",
+				"condition1   1 minute ago",
+				"condition2   20 seconds ago",
+				"condition3   3 weeks ago",
+				"",
+			},
```
Comment (khrm) [inline pkg/cmd/condition/list_test.go]: Can we have a test for https://github.com/tektoncd/cli/pull/319/files#diff-3b9bcd9007c070bb31b5ef44695b2016R81 also? 
Use PrependReactor to simulate error.

Comment (16yuki0702) [reply pkg/cmd/condition/list_test.go]: @khrm Thank you for your review!!
I just added test case, please check this out.

Comment (16yuki0702) [reply pkg/cmd/condition/list_test.go]: @khrm I think test error occured from outputting error message to Err stream.
In my opinion To fix this, the error message should be returned to caller not immediately output Err stream like below.
```
return fmt.Errorf("failed to list conditions from %s namespace \n%s", p.Namespace(), err)
```

But this is different from other error messages.
ex. https://github.com/tektoncd/cli/blob/master/pkg/cmd/pipeline/list.go#L93

If adding this test case I think we should create other pull request to fix all other files, how do you think 🤔??

**Comment by khrm** [comment]
@16yuki0702 Run ```hack/update-codegen.sh``` from root directory to fix the error.

File: pkg/cmd/condition/list_test.go (hunk 1 above)
Comment (16yuki0702) [reply pkg/cmd/condition/list_test.go]: @khrm 
> @16yuki0702 Run hack/update-codegen.sh from root directory to fix the error.

Oh I totally misunderstood :sweat:
very sorry and thank you for your advise :smiley:

File: pkg/cmd/condition/list.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+func printConditionListObj(w io.Writer, p cli.Params, f *cliopts.PrintFlags) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return err
+	}
+
+	conditions, err := listAllConditions(cs.Tekton, p.Namespace())
+	if err != nil {
+		return err
```
Comment (piyush-garg) [inline pkg/cmd/condition/list.go]: Just minor comment can you add here as well `fmt.Fprintf(s.Err, "Failed to list conditions from %s namespace \n", p.Namespace())`

Comment (piyush-garg) [reply pkg/cmd/condition/list.go]: Also if you can add a test for this scenario will be good

**Comment by piyush-garg** [comment]
Also, there is need to rebase and generate docs again.

File: pkg/cmd/condition/list.go (hunk 2 above)
Comment (16yuki0702) [reply pkg/cmd/condition/list.go]: @piyush-garg Thank you for your review!!
I just fixed it :thumbsup:

Comment (piyush-garg) [reply pkg/cmd/condition/list.go]: Thanks

**Comment by chmouel** [comment]
/lgtm
/approve
/meow

(sorry taking time reviewing this, thanks @16yuki0702 )

**Comment by 16yuki0702** [comment]
@khrm @piyush-garg @chmouel Thank for reviews !!

**Pull request merged**